### PR TITLE
Try pipeline caching

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

**Comment by pakrym** [comment]
Seems to save 1-4mins

**Comment by pakrym** [comment]
@benbp, @weshaggard what do you think?

File: eng/pipelines/templates/jobs/ci.tests.yml [hunk 1]
```diff
@@ -47,6 +47,7 @@ jobs:
         - ${{ each pair in step }}:
             ${{ pair.key }}: ${{ pair.value }}
       - template: /eng/pipelines/templates/steps/install-dotnet.yml
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/ci.tests.yml:49]: Why not just include the caching in the install-dotnet.yml template?

File: eng/pipelines/templates/steps/nuget-cache.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+steps:
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/nuget-cache.yml:1]: Just to connect dots it looks like you copied this pretty closely from https://docs.microsoft.com/en-us/azure/devops/pipelines/release/caching?view=azure-devops#netnuget which looks good to me.

File: eng/pipelines/templates/jobs/ci.mgmt.yml [hunk 3]
```diff
@@ -66,9 +63,6 @@ jobs:
       - template: /eng/pipelines/templates/steps/install-dotnet.yml
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/ci.mgmt.yml:63]: Assuming you don't put the caching in the install-dotnet.yml file we should probably add the cache to the mgmt pipelines as well. Assuming @allenjzhang @m-nash @markcowl agree.

File: eng/pipelines/templates/jobs/ci.tests.yml (hunk 1 above)
Comment (pakrym) [reply eng/pipelines/templates/jobs/ci.tests.yml:49]: My thinking was that sometimes somebody would want to run a more lightweight task using dotnet that doesn't require 1.7gb cache.

**Review by weshaggard — APPROVED** [review]
One comment otherwise the change looks good. Obviously we will need to watch it over the next few weeks to ensure it is doing what we expect.

File: eng/pipelines/templates/jobs/ci.tests.yml (hunk 1 above)
Comment (weshaggard) [reply eng/pipelines/templates/jobs/ci.tests.yml:49]: That is fair but I guess we should look at the performance or we could add an opt-out template parameter for the install step for folks that want to disable the cache.

Comment (pakrym) [reply eng/pipelines/templates/jobs/ci.tests.yml:49]: Moved.

**Comment by benbp** [comment]
I didn't know about this feature! I wonder where else this might be useful.

**Pull request merged**